### PR TITLE
ISPN-10790 JDBCTableManipulation config is not validated

### PR DESCRIPTION
**Pull request by wburns**
https://issues.jboss.org/browse/ISPN-10790

### CODE REVIEW

**Comment by wburns** [comment]
Updated to be a bit cleaner.

**Comment by wburns** [comment]
Looks like it caught some issues :)

**Comment by wburns** [comment]
@ryanemerson @tristantarrant I need your guys' help to ascertain how these validate methods should be invoked. This is definitely not correct, the way I have it looking at the test failures.

File: persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/configuration/JdbcStringBasedStoreConfigurationBuilder.java [hunk 1]
```diff
@@ -99,6 +101,23 @@ public StringTableManipulationConfigurationBuilder table() {
       return table;
    }
 
+   @Override
+   public void validate() {
+      if (connectionFactory != null) {
+         connectionFactory.validate();
+      }
+      // TODO: I don't think this is quite correct...
+      if (attributes.attribute(MANAGE_CONNECTION_FACTORY).get() && (connectionFactory == null || connectionFactory.needsManaging())) {
```
Comment (ryanemerson) [inline persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/configuration/JdbcStringBasedStoreConfigurationBuilder.java]: Can't we do this without the need for the `needsManaging` interface. The following should be sufficient:

```java
if (!attributes.attribute(MANAGE_CONNECTION_FACTORY).get())
   table.validate()
```

If we add `builder.manageConnectionFactory(false);` to the `ManagedConnectionFactoryConfigurationBuilder` constructor.

**Comment by ryanemerson** [comment]
@wburns I noticed that BATCH_SIZE is marked for removal in 10.0. Can you add a commit to this PR removing it? I don't want to create another PR as it'll conflict with this probably

**Comment by wburns** [comment]
Updated to revert back to always requiring the fields (except segments only if segmented).

**Comment by ryanemerson** [comment]
@wburns Looks good on CI, just needs the commits squashing

**Comment by wburns** [comment]
Squashed to one commit.

**Pull request merged**